libraries/AP_RangeFinder/RangeFinder_Backend.h

```diff
@@ -80,8 +79,12 @@ class AP_RangeFinder_Backend
     void set_status(RangeFinder::RangeFinder_Status status);
 
     RangeFinder::RangeFinder_State &state;
-    MAV_DISTANCE_SENSOR sensor_type;
 
     // semaphore for access to shared frontend data
     AP_HAL::Semaphore *_sem;    
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const {
+        // most of the backends are lasers:
+        return MAV_DISTANCE_SENSOR_LASER;
```

Comment:
@OXINARF I like the pure-virtual approach.  Only MAVLink doesn't know ahead of time what the hardware is, and it is happily doing its own thing.